ui/core/components/suggest_reforges_action.tsx

```diff
@@ -650,7 +649,9 @@ export class ReforgeOptimizer {
 			console.log('The following slots will not be cleared:');
 			console.log(Array.from(this.frozenItemSlots.keys()).filter(key => this.frozenItemSlots.get(key)));
 		}
-		const baseGear = this.player.getGear().withoutReforges(this.player.canDualWield2H(), this.frozenItemSlots);
+		const previousGear = this.player.getGear();
+		this.previousGear = previousGear.getEquippedItems();
```

Comment:
Then you could call `this.previousReforges = previousGear.getAllReforges();`. Where you'd have a new `Gear` method defined to loop through all item slots with a non-null `EquippedItem` and a non-null value for `equippedItem.reforge`, and call `reforgeMap.set(slot, equippedItem.reforge)`. This returns a clean map without any nulls to deal with.